File: client/app-ui/src/bookTypes.ts
export type BookTypeConfig = {
  id: string
  shortLabel: string
  displayName: string
  description: string
  recommendedMode: 'classical' | 'expert'
  allowedModes: Array<'classical' | 'expert'>
  discouragedModes?: Array<'classical' | 'expert'>
  hardDisabledModes?: Array<'classical' | 'expert'>
  defaultSkill: string
  sourcePolicy: 'optional' | 'recommended' | 'required'
  defaultFormat: 'classic-novel' | 'illustrated-story' | 'modern-editorial'
  defaultTone: string
}

export const BOOK_TYPES: BookTypeConfig[] = [
  { id: 'fiction_novel', shortLabel: 'Novel', displayName: 'Fiction Novel', description: 'Long-form narrative fiction, scenes and characters.', recommendedMode: 'classical', allowedModes: ['classical', 'expert'], defaultSkill: 'fiction_book_page', sourcePolicy: 'optional', defaultFormat: 'classic-novel', defaultTone: 'immersive, literary, scene-driven' },
  { id: 'memoir_personal_story', shortLabel: 'Memoir', displayName: 'Memoir / Personal Story', description: 'Personal narrative storytelling.', recommendedMode: 'classical', allowedModes: ['classical', 'expert'], defaultSkill: 'fiction_book_page', sourcePolicy: 'optional', defaultFormat: 'classic-novel', defaultTone: 'reflective, vivid, honest' },
  { id: 'childrens_illustrated_book', shortLabel: 'Illustrated', displayName: 'Children’s Illustrated Book', description: 'Short, image-led story pages.', recommendedMode: 'classical', allowedModes: ['classical', 'expert'], defaultSkill: 'fiction_book_page', sourcePolicy: 'optional', defaultFormat: 'illustrated-story', defaultTone: 'warm and clear' },
  { id: 'marketing_story', shortLabel: 'Marketing', displayName: 'Marketing Story / Brand Book', description: 'Professional GTM and brand narrative.', recommendedMode: 'expert', allowedModes: ['expert', 'classical'], defaultSkill: 'marketing_book_page', sourcePolicy: 'recommended', defaultFormat: 'modern-editorial', defaultTone: 'professional, audience-aware, evidence-led' },
  { id: 'finance_explainer', shortLabel: 'Finance', displayName: 'Finance Explainer', description: 'Practical finance guidance and frameworks.', recommendedMode: 'expert', allowedModes: ['expert', 'classical'], defaultSkill: 'finance_book_page', sourcePolicy: 'recommended', defaultFormat: 'modern-editorial', defaultTone: 'precise, practical, careful' },
  { id: 'thought_leadership', shortLabel: 'Leadership', displayName: 'Business Thought Leadership', description: 'POV content for business leaders.', recommendedMode: 'expert', allowedModes: ['expert', 'classical'], defaultSkill: 'general_book_page', sourcePolicy: 'recommended', defaultFormat: 'modern-editorial', defaultTone: 'insightful and practical' },
  { id: 'case_study_collection', shortLabel: 'Case Study', displayName: 'Case Study Collection', description: 'Evidence-led case narratives.', recommendedMode: 'expert', allowedModes: ['expert', 'classical'], discouragedModes: ['classical'], defaultSkill: 'marketing_book_page', sourcePolicy: 'required', defaultFormat: 'modern-editorial', defaultTone: 'clear and concrete' },
  { id: 'educational_how_to', shortLabel: 'Guide', displayName: 'Educational / How-To Book', description: 'Instructional and learning-focused.', recommendedMode: 'classical', allowedModes: ['classical', 'expert'], defaultSkill: 'general_book_page', sourcePolicy: 'optional', defaultFormat: 'classic-novel', defaultTone: 'clear and instructive' },
  { id: 'custom', shortLabel: 'Custom', displayName: 'Custom', description: 'Custom objective and tone.', recommendedMode: 'classical', allowedModes: ['classical', 'expert'], defaultSkill: 'general_book_page', sourcePolicy: 'optional', defaultFormat: 'classic-novel', defaultTone: 'balanced' },
]

export const BOOK_TYPE_MAP = Object.fromEntries(BOOK_TYPES.map((b) => [b.id, b])) as Record<string, BookTypeConfig>

File: server/app/core/book_types.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class BookTypeConfig:
    id: str
    short_label: str
    display_name: str
    description: str
    recommended_mode: str
    allowed_modes: tuple[str, ...]
    default_skill: str
    source_policy: str
    default_format: str
    default_tone: str
    discouraged_modes: tuple[str, ...] = ()
    hard_disabled_modes: tuple[str, ...] = ()


BOOK_TYPES: dict[str, BookTypeConfig] = {
    "fiction_novel": BookTypeConfig("fiction_novel", "Novel", "Fiction Novel", "Long-form narrative fiction.", "classical", ("classical", "expert"), default_skill="fiction_book_page", source_policy="optional", default_format="classic-novel", default_tone="immersive, literary, scene-driven"),
    "memoir_personal_story": BookTypeConfig("memoir_personal_story", "Memoir", "Memoir / Personal Story", "Personal narrative nonfiction.", "classical", ("classical", "expert"), default_skill="fiction_book_page", source_policy="optional", default_format="classic-novel", default_tone="reflective, vivid, honest"),
    "childrens_illustrated_book": BookTypeConfig("childrens_illustrated_book", "Illustrated", "Children’s Illustrated Book", "Short image-led story pages.", "classical", ("classical", "expert"), default_skill="fiction_book_page", source_policy="optional", default_format="illustrated-story", default_tone="warm, playful, clear"),
    "marketing_story": BookTypeConfig("marketing_story", "Marketing", "Marketing Story / Brand Book", "Audience-aware business narrative.", "expert", ("expert", "classical"), default_skill="marketing_book_page", source_policy="recommended", default_format="modern-editorial", default_tone="professional, audience-aware, evidence-led"),
    "finance_explainer": BookTypeConfig("finance_explainer", "Finance", "Finance Explainer", "Practical finance clarity.", "expert", ("expert", "classical"), default_skill="finance_book_page", source_policy="recommended", default_format="modern-editorial", default_tone="precise, practical, careful"),
    "thought_leadership": BookTypeConfig("thought_leadership", "Leadership", "Business Thought Leadership", "Point-of-view business writing.", "expert", ("expert", "classical"), default_skill="general_book_page", source_policy="recommended", default_format="modern-editorial", default_tone="insightful, structured, practical"),
    "case_study_collection": BookTypeConfig("case_study_collection", "Case Study", "Case Study Collection", "Evidence-led case compendium.", "expert", ("expert", "classical"), discouraged_modes=("classical",), default_skill="marketing_book_page", source_policy="required", default_format="modern-editorial", default_tone="clear, practical, concrete"),
    "educational_how_to": BookTypeConfig("educational_how_to", "Guide", "Educational / How-To Book", "Step-by-step practical teaching.", "classical", ("classical", "expert"), default_skill="general_book_page", source_policy="optional", default_format="classic-novel", default_tone="clear, instructive, practical"),
    "custom": BookTypeConfig("custom", "Custom", "Custom", "Custom purpose and style.", "classical", ("classical", "expert"), default_skill="general_book_page", source_policy="optional", default_format="classic-novel", default_tone="balanced"),
}


def get_book_type_config(book_type_id: str | None) -> BookTypeConfig:
    if book_type_id and book_type_id in BOOK_TYPES:
        return BOOK_TYPES[book_type_id]
    return BOOK_TYPES["custom"]
